### ISSUE TITLE
Error: Link doesn't refer to the right username, expected: zQueal got: zqueal

### ISSUE THREAD

**Issue by zQueal** [keybase/keybase-issues#1654]
```
λ keybase -d login
debug:  Fallback to layout_v1
debug: + opening config file: C:\Users\zqueal\AppData\Roaming\keybase\config.json
debug: ++ loading config file C:\Users\zqueal\AppData\Roaming\keybase\config.json
debug: -- loaded config file -> {"user":{"email":"zach.queal@gmail.com","salt":"144f152ccb400696a4bde620cff82883","id":"15b4292135d94eba7afb70290fdcc900","name":"zqueal"}}
debug: - opened config file; found=true
debug: + testing GPG command-line client <default: gpg2 or gpg>
debug: | using GPG command: gpg2
debug: - tested GPG command-line client -> null
debug: Warning on pinentry init: Error: can only run on Linux and OSX
debug: + Load proxy CAs
debug: - Loaded proxy CAs
debug: + startup message
debug: | CLI version: 0.8.4
debug: | Platform info: {"versions":{"http_parser":"2.3","node":"0.12.2","v8":"3.28.73","uv":"1.4.2-node1","zlib":"1.2.8","modules":"14","openssl":"1.0.1m"},"arch":"ia32","platform":"win32","features":{"debug":false,"uv":true,"ipv6":true,"tls_npn":true,"tls_sni":true,"tls_ocsp":true,"tls":true}}
debug: | Version info: ["keybase (keybase.io CLI) v0.8.4","- node.js v0.12.2","- gpg (GnuPG) 2.0.27 (Gpg4win 2.2.4)\r","- libgcrypt 1.6.3\r","Identifies as: 'keybase.io node.js client v0.8.4 win32'"]
debug: - startup message
debug: + opening database file: C:\Users\zqueal\AppData\Roaming\keybase\keybase.idb
debug: ++ DB::_init_db
debug: -- DB::_init_db -> 0
debug: - DB opened
debug: + session::load
debug: ++ opening config file: C:\Users\zqueal\AppData\Roaming\keybase\session.json
debug: +++ loading config file C:\Users\zqueal\AppData\Roaming\keybase\session.json
debug: --- loaded config file -> <redacted>
debug: -- opened config file; found=true
debug: - session::load
debug: + session::check
debug: | calling to sesscheck
debug: ++ request to sesscheck (https://api.keybase.io:443/_/api/1.0/sesscheck.json) (cookie=true)
debug: || Adding a custom CA for host api.keybase.io when tls=true
debug: -- request to sesscheck -> null
debug: - session::check
debug: + KeyPull::run
debug: ++ KeyPull::load_user
debug: +++ zqueal: load user
debug: ++++ zqueal: load user from local storage
debug: +++++ load sig chain from local storage
debug: ||||| No payload hash tail pointer found
debug: ----- loaded sig chain from local storage
debug: ---- zqueal: loaded user from local storage -> null / [object Object]
debug: ||| Checking session since we're loading User as self (and need secret key)
debug: ++++ session::load_and_check
debug: +++++ session::check
debug: ||||| calling to sesscheck
debug: ++++++ request to sesscheck (https://api.keybase.io:443/_/api/1.0/sesscheck.json) (cookie=true)
debug: |||||| Adding a custom CA for host api.keybase.io when tls=true
debug: ------ request to sesscheck -> null
debug: ----- session::check
debug: ---- session::load_and_check -> true
debug: ++++ zqueal: load user from server
debug: +++++ merkle find_and_verify: uid undefined, username zqueal
debug: ++++++ request to merkle/path (https://api.keybase.io:443/_/api/1.0/merkle/path.json?uid=&username=zqueal) (cookie=true)
debug: |||||| Adding a custom CA for host api.keybase.io when tls=true
debug: ------ request to merkle/path -> null
debug: ++++++ merkle get_merkle_pgp_key
debug: |||||| Call to gpg2: { args: [ '-k', '--with-fingerprint', '--with-colons' ],
debug:   quiet: true,
debug:   no_options: true }
debug: |||||| merkle key already found in keyring
debug: ------ merkle get_merkle_pgp_key
debug: ||||| Load key /2A32340CEC8C9492 from master keyring (secret=undefined)
debug: ||||| Call to gpg2: { args:
debug:    [ '--export',
debug:      '--export-options',
debug:      'export-local-sigs',
debug:      '-a',
debug:      '2A32340CEC8C9492' ] }
debug: ++++++ lookup UID
debug: +++++++ read_uids_from_keys 03E146CDAF8136680AD566912A32340CEC8C9492
debug: ||||||| Call to gpg2: { args:
debug:    [ '-k',
debug:      '--with-fingerprint',
debug:      '--with-colons',
debug:      '03E146CDAF8136680AD566912A32340CEC8C9492' ],
debug:   quiet: true,
debug:   no_options: true }
debug: ||||||| got: [{"username":"Keybase.io Merkle Signing","comment":"v1","email":"merkle@keybase.io"}]
debug: ------- read_uids_from_key -> null
debug:  - Map 2A32340CEC8C9492 -> [object Object] via gpg
debug: ------ looked up UID
debug: ++++++ Rollback check
debug: |||||| DB put value kv:e2:8aa751c53d6279a39feadb333999099d44a76c2e823e3488572bf3d46e4052664369e0c4a6ccfe6be86a4c0448319f4162c2a971d8520a3121fd8b8e21956f08
debug: |||||| DB put lookup: lo:5:latest -> d7addb131e95f98b4dd1ad6db331d5c9327ea1395239fc9bfb4f3844e2c39339
debug: ------ Rollback check
debug: ++++++ request to user/lookup (https://api.keybase.io:443/_/api/1.0/user/lookup.json?username=zqueal) (cookie=true)
debug: |||||| Adding a custom CA for host api.keybase.io when tls=true
debug: ------ request to user/lookup -> null
debug: ----- zqueal: loaded user from server
debug: +++++ updating local user w/ remote
debug: ++++++ update sig chain; remote_seqno=70
debug: |||||| sigchain update: 70 vs. null
debug: +++++++ sigchain::_update
debug: ++++++++ request to sig/get (https://api.keybase.io:443/_/api/1.0/sig/get.json?uid=15b4292135d94eba7afb70290fdcc900&low=1) (cookie=true)
debug: |||||||| Adding a custom CA for host api.keybase.io when tls=true
debug: -------- request to sig/get -> null
debug: ||||||| found 70 new signatures
debug: ------- sigchain::_update
debug: |||||| update sig_chain last link to {"seqno":70,"payload_hash":"9fdc4a1617c474ace33c1e40da8f26e4d4acf25f6de542fca37f520f807d5665","sig_id":"d6216f68395f5ec331a14cee0ce2ce6e5f4c96a504b0aa84c6227f521e48762e0f"}
debug: ------ updated sig chain
debug: ----- finished update
debug: +++++ zqueal: verifying signatures
debug: ++++++ zqueal: verifying sig
error: Link doesn't refer to the right username, expected: zQueal got: zqueal
```

Really interesting bug. I just updated Keybase and ran into this. Was working just fine before.

```
λ whoami
xanza-pc\zqueal
```

**Comment by oconnor663** [comment]
Woops, thanks for the report. I just rewrote some of our sigchain parsing, and I forgot that we used to allow non-lowercase usernames. Should be an easy fix.

**Comment by maxtaco** [comment]
Fixed in v0.8.5, sorry about the bug, and thanks for the bug report!
